### DISCUSSION TITLE
Actor Restart

### DISCUSSION THREAD

**Discussion by HermansG** [akkadotnet/akka.net#7417] [Q&A] [answered]
When an actor crashes, it can be in a certain State e.g. state C. 
Perhaps the actor went from default State A to state B then to state C.
When the actor is restarted, it will be back in its default state A. True?
In the video about Actor Restart this is not mentioned.
I suppose you shouldn't make Actors too complicated.

**Comment by Aaronontheweb** [comment, marked as answer]
> When the actor is restarted, it will be back in its default state A. True?

That is correct - however, [persistent actors](https://getakka.net/articles/persistence/event-sourcing.html) have the ability to recover their last state by replaying the latest snapshot (optional) and events. This can be used to ensure that the actor reboots into state C, rather than A.

You don't necessarily _have_ to have Akka.Persistence to achieve this type of re-entrancy - I'm using some EF Core repositories in one of our legacy applications to allow actors to recover their most recent state and I'm just using a bit of manual behavior-switching to mimic what Akka.Persistence does. But the idea remains the same: give actors the ability to recover their own previous state upon restart under their own power, without relying on external messaging to do so.

**Reply by HermansG** [reply]
Persistent actor behaviour has no equivalence in OOP has it?
Very clever to be able to also recover a State, still I suppose I think it is too complicated to go in that direction.

**Reply by Aaronontheweb** [reply]
For an example of recovering state without Akka.Persistence, our latest video on "[Deferred Actor Message Processing with Stashing](https://www.youtube.com/watch?v=S8FNHZ7SES8)" has examples on how to recover state without persistence.
